test_pull_requests

```diff
@@ -1583,30 +1564,49 @@ popd
       end
     end
 
-    def self.to_api_url build_url
-      URI(build_url + "/api/json")
-    end
-
-    def self.is_previous_build_running?(branch, settings)
-      build_num = get_next_build(branch, settings) - 1
-      build_url = Properties['jenkins_url'] + 'job/' + branch_settings(branch, settings)['jenkins_job_name'] + "/#{build_num}/"
+    def self.previous_build_running?(branch, settings, repo)
+      addtl_jenkins_params = branch_settings(branch, settings)['addtl_jenkins_params']
+      previous_running = false
       get_session(branch, settings).start do |http|
-        request = Net::HTTP::Get.new to_api_url(build_url).request_uri
-        response = http.request request
+        build_uri = URI(Properties['jenkins_url'] + 'job/' + branch_settings(branch, settings)['jenkins_job_name'] + 'api/json??tree=builds[building,actions[parameters[name,value]]]')
+        response = http_get(build_url)
         if response.kind_of? Net::HTTPSuccess
-          return JSON.parse(response.body)["building"]
+          builds = JSON.parse(response.body)
+          builds.each do |build|
```

Comment:
also need to do wildcard stuff, etc, clearly not a full solution but a suggestion